### PR TITLE
Unify FormulaType Json representation with other type definitions

### PR DESCRIPTION
**Pull request by lesaltzm**

### CODE REVIEW

File: src/libraries/Microsoft.PowerFx.Core/Public/Types/Serialization/FormulaTypeSchema.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        
+        /// <summary>
+        /// Optional. Used for external schema definitions and input validation.
+        /// </summary>
+        public bool? Required { get; set; }
+
+        /// <summary>
+        /// Optional. For entities, specifies the table logical name.
+        /// </summary>
+        public string TableLogicalName { get; set; }
```
Comment (rsullivan00) [inline src/libraries/Microsoft.PowerFx.Core/Public/Types/Serialization/FormulaTypeSchema.cs:51]: This looks unused. Just checking whether you want this in this PR or added later w/ entity support.

**Review by rsullivan00 — APPROVED** [review]

File: src/libraries/Microsoft.PowerFx.Core/Public/Types/Serialization/FormulaTypeToSchemaConverter.cs [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+                Result = new FormulaTypeSchema()
+                {
+                    Type = FormulaTypeSchema.ParamType.Table,
+                    Fields = GetChildren(type)
+                };
+            }
+
+            private Dictionary<string, FormulaTypeSchema> GetChildren(AggregateType type)
+            {
+                var fields = new Dictionary<string, FormulaTypeSchema>();
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Public/Types/Serialization/FormulaTypeToSchemaConverter.cs:117]: >(); [](http://example.com/codeflow?start=70&length=3)

case sensitive?

**Review by MikeStall — APPROVED** [review]
:shipit:

File: src/libraries/Microsoft.PowerFx.Core/Public/Types/Serialization/FormulaTypeSchema.cs (hunk 1 above)
Comment (lesaltzm) [reply src/libraries/Microsoft.PowerFx.Core/Public/Types/Serialization/FormulaTypeSchema.cs:51]: Unused by standalone PowerFx, but used by canvas, as seen here https://github.com/microsoft/PowerApps-Language-Tooling/blob/master/src/PAModel/Schemas/ParameterSchema.cs

**Review by MikeStall — APPROVED** [review]

**Pull request merged**